api/v1alpha1/gateway_types.go

```diff
@@ -285,14 +289,21 @@ const (
 	NamedAddressType AddressType = "NamedAddress"
 )
 
-// ListenerAddress describes an address for the Listener.
-type ListenerAddress struct {
-	// Type of the Address. This is one of the *AddressType constants.
+// GatewayAddress describes an address that can be bound to a Gateway.
+type GatewayAddress struct {
+	// Type of the Address. This is one of the AddressType constants.
```

Comment:
Since there can only be 2 possible values here, it might make sense to list them in the comments unless the documentation can pick up the annotations.